### PR TITLE
Add staging prod environments

### PR DESCRIPTION
**Pull request by kmahern**
#### What
Create prod and staging environments

#### Ticket
https://dsdmoj.atlassian.net/browse/CACP-126
https://dsdmoj.atlassian.net/browse/CACP-127

#### Why
In order to continue development and eventually deploy to a production environment

#### How

--------

#### TODO (wip)

 - [ ] item 1
 - [X] item 2

### CODE REVIEW

File: .k8s/prod/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: laa-court-data-ui
+  namespace: laa-court-data-ui-prod
```
Comment (jsugarman) [inline .k8s/prod/deployment.yaml]: do no abbreviate  - `production`

File: .k8s/prod/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: laa-court-data-ui-app-ingress
+  namespace: laa-court-data-ui-prod
```
Comment (jsugarman) [inline .k8s/prod/ingress.yaml]: do no abbreviate  - `production`

File: .k8s/prod/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      http:
+        paths:
+          - path: /
+            backend:
+              serviceName: laa-court-data-ui-app-service
+              servicePort: 80
+  tls:
+    - hosts:
+      - prod.view-court-data.service.justice.gov.uk
+      secretName: laa-court-data-ui-prod-cert
```
Comment (jsugarman) [inline .k8s/prod/ingress.yaml]: do no abbreviate  - `production`

File: .k8s/prod/service.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: laa-court-data-ui-app-service
+  namespace: laa-court-data-ui-prod
```
Comment (jsugarman) [inline .k8s/prod/service.yaml]: do no abbreviate  - `production`

File: .k8s/scripts/deploy.sh [hunk 5]
```diff
@@ -26,7 +26,7 @@ function _deploy() {
   fi
 
   case "$1" in
-    dev)
+    dev | staging | prod)
```
Comment (jsugarman) [inline .k8s/scripts/deploy.sh]: do no abbreviate  - `production`

File: .k8s/prod/ingress.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    - host: prod.view-court-data.service.justice.gov.uk
+      http:
+        paths:
+          - path: /
+            backend:
+              serviceName: laa-court-data-ui-app-service
+              servicePort: 80
+  tls:
+    - hosts:
+      - prod.view-court-data.service.justice.gov.uk
```
Comment (jsugarman) [inline .k8s/prod/ingress.yaml]: this should just be `view-court-data.service.justice.gov.uk`

File: .k8s/prod/ingress.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: laa-court-data-ui-app-ingress
+  namespace: laa-court-data-ui-prod
+spec:
+  rules:
+    - host: prod.view-court-data.service.justice.gov.uk
```
Comment (jsugarman) [inline .k8s/prod/ingress.yaml]: this should just be `view-court-data.service.justice.gov.uk`

File: .k8s/staging/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,90 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: laa-court-data-ui
+  namespace: laa-court-data-ui-staging
+spec:
+  replicas: 2
```
Comment (jsugarman) [inline .k8s/staging/deployment.yaml]: lets have 4 replicas

File: .k8s/prod/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+spec:
+  replicas: 2
```
Comment (jsugarman) [inline .k8s/prod/deployment.yaml]: lets have 4 replicas

File: .k8s/prod/deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+              httpHeaders:
+                - name: X-Forwarded-Proto
+                  value: https
+                - name: X-Forwarded-Ssl
+                  value: "on"
+            initialDelaySeconds: 30
+            periodSeconds: 10
+          env:
+            - name: ENV
+              value: 'prod'
```
Comment (jsugarman) [inline .k8s/prod/deployment.yaml]: do not abbreviate - `production`

File: .k8s/prod/deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+            - name: RACK_ENV
+              value: 'production'
```
Comment (jsugarman) [inline .k8s/prod/deployment.yaml]: this should match `RAILS_ENV`

File: .k8s/prod/deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+            - name: RAILS_ENV
+              value: 'production'
+            - name: DOMAIN_URL
+              value: 'https://prod.view-court-data.service.justice.gov.uk'
```
Comment (jsugarman) [inline .k8s/prod/deployment.yaml]: should be https://view-court-data.service.justice.gov.uk

File: .k8s/prod/deployment.yaml (hunk 9 above)
Comment (kmahern) [reply .k8s/prod/deployment.yaml]: 👍

File: .k8s/staging/deployment.yaml (hunk 8 above)
Comment (kmahern) [reply .k8s/staging/deployment.yaml]: 👍

File: .k8s/prod/ingress.yaml (hunk 7 above)
Comment (kmahern) [reply .k8s/prod/ingress.yaml]: 👍

File: .k8s/prod/ingress.yaml (hunk 6 above)
Comment (kmahern) [reply .k8s/prod/ingress.yaml]: 👍

File: .k8s/scripts/deploy.sh (hunk 5 above)
Comment (kmahern) [reply .k8s/scripts/deploy.sh]: 👍

File: .k8s/prod/service.yaml (hunk 4 above)
Comment (kmahern) [reply .k8s/prod/service.yaml]: 👍

File: .k8s/prod/ingress.yaml (hunk 3 above)
Comment (kmahern) [reply .k8s/prod/ingress.yaml]: 👍

File: .k8s/prod/ingress.yaml (hunk 2 above)
Comment (kmahern) [reply .k8s/prod/ingress.yaml]: 👍

File: .k8s/prod/deployment.yaml (hunk 10 above)
Comment (kmahern) [reply .k8s/prod/deployment.yaml]: 👍

File: .k8s/prod/deployment.yaml (hunk 11 above)
Comment (kmahern) [reply .k8s/prod/deployment.yaml]: 👍

File: .k8s/prod/deployment.yaml (hunk 12 above)
Comment (kmahern) [reply .k8s/prod/deployment.yaml]: 👍

File: .k8s/prod/deployment.yaml (hunk 1 above)
Comment (kmahern) [reply .k8s/prod/deployment.yaml]: 👍

File: .k8s/production/deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+            - name: RAILS_ENV
+              value: 'production'
+            - name: DOMAIN_URL
+              value: 'https://view-court-data.service.justice.gov.uk'
+            - name: COURT_DATA_ADAPTOR_API_URL
+              value: https://laa-court-data-adaptor-prod.apps.live-1.cloud-platform.service.justice.gov.uk/api/internal/v1
+            - name: RAILS_SERVE_STATIC_FILES
+              value: enabled
+            - name: GA_TRACKING_ID
+              value: 'UA-131160087-14'
```
Comment (jsugarman) [inline .k8s/production/deployment.yaml]: The production and staging namespaces/environments will need their own google analytics tracking code

File: .k8s/staging/deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+            - name: RAILS_ENV
+              value: 'production'
+            - name: DOMAIN_URL
+              value: 'https://staging.view-court-data.service.justice.gov.uk'
+            - name: COURT_DATA_ADAPTOR_API_URL
+              value: https://laa-court-data-adaptor-staging.apps.live-1.cloud-platform.service.justice.gov.uk/api/internal/v1
+            - name: RAILS_SERVE_STATIC_FILES
+              value: enabled
+            - name: GA_TRACKING_ID
+              value: 'UA-131160087-14'
```
Comment (jsugarman) [inline .k8s/staging/deployment.yaml]: The production and staging namespaces/environments will need their own google analytics tracking code. 

ask @naseberry to help you out with that

File: .k8s/staging/deployment.yaml [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+            - name: ENV
+              value: 'staging'
+            - name: RACK_ENV
+              value: 'production'
+            - name: RAILS_ENV
+              value: 'production'
+            - name: DOMAIN_URL
+              value: 'https://staging.view-court-data.service.justice.gov.uk'
+            - name: COURT_DATA_ADAPTOR_API_URL
+              value: https://laa-court-data-adaptor-staging.apps.live-1.cloud-platform.service.justice.gov.uk/api/internal/v1
```
Comment (jsugarman) [inline .k8s/staging/deployment.yaml]: unfortunately this is called 

https://laa-court-data-adaptor-stage.apps.live-1.cloud-platform.service.justice.gov.uk/api/internal/v1

`staging --> stage`

Comment (kmahern) [reply .k8s/staging/deployment.yaml]: 👍

**Pull request merged**